portablemc/optifine.py

```diff
@@ -0,0 +1,659 @@
+import re # needed to parse downloads from optifine website and parse the patch config
+import zipfile
+# Libs for binary files processing
+from io import BytesIO
+import struct
+from typing import List, Dict, Optional
+from datetime import datetime
+from hashlib import md5 ,sha1# check if the file is corrupted after the patch
+from .standard import Version, Watcher, Context, VersionHandle, VersionNotFoundError
+from .http import http_request, HttpError
+from .download import DownloadEntry
+from .util import LibrarySpecifier
+from pathlib import Path
+
+class OptifineEntryObject:
+    def __init__(self, filename: str, edition: str, mc_version: str, preview: bool =False, date: Optional[str] | None =None, forge: Optional[str] | None =None):
```

Comment:
Type annotation `Optional[T] | None` is redundant, just use `Optional[T]`, the `| None` syntax is only supported since Python 3.10 and we want to support down to 3.8